### PR TITLE
adding mobile left menu [MDS-216]

### PR DESCRIPTION
**Pull request by phgrey**
left menu for mobile version of the site

### CODE REVIEW

File: lib/moon_web/components/header.ex [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  use MoonWeb, :stateless_component
+
+  prop click, :event
+  slot default
+
+  def render(assigns) do
+    ~F"""
+    <header>
+      <div class="relative z-10 py-4 lg:hidden flex flex-row gap-2 items-center justify-between text-bulma">
+        <button type="button" class="focus:outline-none" phx-click={@click}>
```
Comment (margusp2heathmont) [inline lib/moon_web/components/header.ex:13]: Should it be :on-click otherwise target might be wrong?

File: lib/moon_web/components/header.ex [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        </button>
+        <a href="/">
+          <svg
+            width="1em"
+            height="1em"
+            viewBox="0 0 24 24"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+            font-size="2rem"
+            class="sc-eqhNQH fcnWEQ"
```
Comment (margusp2heathmont) [inline lib/moon_web/components/header.ex:34]: Why SVG is inline not in separate file?

File: lib/moon_web/components/left_menu.ex [hunk 3]
```diff
@@ -15,13 +15,19 @@ defmodule MoonWeb.Components.LeftMenu do
...
+  prop click, :event
 
   def render(assigns) do
     ~F"""
-    <div class={
-      "hidden fixed h-screen lg:flex lg:flex-shrink-0 w-80 flex-col",
-      @theme_name
-    }>
+    <div
+      phx-click={@click}
```
Comment (margusp2heathmont) [inline lib/moon_web/components/left_menu.ex:23]: :on-click is for surface

File: lib/moon_web/components/header.ex (hunk 1 above)
Comment (phgrey) [reply lib/moon_web/components/header.ex:13]: i'm using LiveView.JS events here, simple DOM manipulation. so target is parent liveView, and that's what's needed here bc "#left-menu-container" is outside Header component.

File: lib/moon_web/components/left_menu.ex (hunk 3 above)
Comment (phgrey) [reply lib/moon_web/components/left_menu.ex:23]: the same - LiveView event is used to set target to the parent LiveView

File: lib/moon_web/components/header.ex (hunk 2 above)
Comment (phgrey) [reply lib/moon_web/components/header.ex:34]: i need to pass a connection/socket property  through 3 levels to get static path for theese svgs. looks like it's more complicated. what do you think?

Comment (phgrey) [reply lib/moon_web/components/header.ex:34]: is there a simlier way to get static path than `{static_path(@socket, "/svgs/moon_web/moon_logo.svg")}` ?

File: lib/moon_web/components/left_menu.ex (hunk 3 above)
Comment (margusp2heathmont) [reply lib/moon_web/components/left_menu.ex:23]: If someone starts using this component from other component, then automatic behaviour might be unexpected. Let's try to use :on-click

File: lib/moon_web/pages/getting_started_page.ex [hunk 4]
```diff
@@ -39,7 +38,7 @@ defmodule MoonWeb.Pages.GettingStartedPage do
           class="p-0.5 rounded-lg flex bg-beerus-100"
         />
       </div>
-      <ForDesigner :if={@selected_role == List.first(@roles)} socket={@socket} />
+      <ForDesigner :if={@selected_role == List.first(@roles)} />
```
Comment (margusp2heathmont) [inline lib/moon_web/pages/getting_started_page.ex:41]: Why this does not check:

`@selected_role == "I'm a designer"`

or 

`@selected_role == @designer_role`

Too complex?

**Review by margusp2heathmont — CHANGES_REQUESTED** [review]
Good work! One small change pls.

File: lib/moon_web/pages/getting_started_page.ex (hunk 4 above)
Comment (phgrey) [reply lib/moon_web/pages/getting_started_page.ex]: done, but with functions - they're available boith in mount() and render()

File: lib/moon_web/pages/getting_started_page.ex [hunk 5]
```diff
@@ -33,19 +31,22 @@ defmodule MoonWeb.Pages.GettingStartedPage do
...
+      <ForDeveloper :if={@selected_role == developer_role()} />
     </Page>
     """
   end
 
   def handle_event("handle_role", %{"selected-item" => selected_role}, socket) do
     {:noreply, assign(socket, :selected_role, selected_role)}
   end
+
+  defp designer_role(), do: "I'm a designer"
```
Comment (margusp2heathmont) [inline lib/moon_web/pages/getting_started_page.ex:50]: Why 1 function is def and other is defp? 
Why it needs to be a function but not module variable for example?

Comment (phgrey) [reply lib/moon_web/pages/getting_started_page.ex:50]: developer_role is used in for_developer.ex , tha's why it's public
functions can be used both in template & mount function, and we do need theese values there

Comment (phgrey) [reply lib/moon_web/pages/getting_started_page.ex:50]: and I couldn't find a way to use module variable in template

**Review by margusp2heathmont — APPROVED** [review]

**Pull request merged**